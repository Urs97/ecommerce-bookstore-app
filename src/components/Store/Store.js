import React, { useEffect, useRef, useContext } from 'react';
import './Store.scss';

import StoreContext from '../../context/StoreContext';
import BooksContainer from '../BooksContainer/BooksContainer';
import StoreSidebar from '../StoreSidebar/StoreSidebar';
import PageHeader from '../PageHeader/PageHeader';
import Pagination from '../Pagination/Pagination';

const Store = () => {
    const context = useContext(StoreContext);
    const storeMainRef = useRef(null);

    useEffect(() => {
        (context.state.storeScrollState > 0 && 
            window.scrollTo(0, storeMainRef.current.offsetTop));
    }, [context.state.storeScrollState])

    let content = (
        <main>
            <PageHeader title="Loading Resources..." color="blue" />
            <section className="loader" />
        </main>);

    if(!context.isLoading && context.data) {
        content = (
            <main>
                <PageHeader title={context.data.name} color="blue" />
                <section className="store-main-container" ref={storeMainRef}>
                    <section className="books-container-pagination-wrapper">
                        <BooksContainer bookData={context.currentBooks} />
                        <Pagination />
                    </section>
                    <StoreSidebar />
                </section>
            </main>
        )
        
    } else if (!context.isLoading && context.data === null) {
        content = (
            <main>
                <PageHeader title={"Failed to load"} color="blue" />
                <section className="failed-to-load-store">
                    <h3 className="title">Failed to load resources.</h3>
                    <p>Please check your internet connection and reload the page.</p>
                </section>
            </main>
        )
    }
    return content;
};

export default Store;