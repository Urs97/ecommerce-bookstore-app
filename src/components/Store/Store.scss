@import '../../styles/shared.scss';

.store-main-container {
    display: grid;
    grid-template-columns: 7fr 3fr;

    padding: $vertical-content-padding $horizontal-content-padding;
    background-color: $background-grey;

    .books-container-pagination-wrapper {
        & .books-container-section {
            padding: 0;
    
            .books {
                gap: 3em;
            }
        }
    }
}

.loader {
    border: 1em solid #D3D3D3; 
    border-top: 1em solid #313131; 
    border-radius: 50%;
    width: 7.5em;
    height: 7.5em;
    animation: spin 2s linear infinite;
    margin: 0 auto;
    margin-top: 7em;
    margin-bottom: 10em;
}

@keyframes spin {
    0% { transform: rotate(0deg) }
    100% { transform: rotate(360deg) }
}