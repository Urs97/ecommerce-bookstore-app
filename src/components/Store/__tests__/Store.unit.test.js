import React from "react";
import { shallow } from "enzyme";
import Store from "../Store";

describe("Store", () => {
  test("renders Store", () => {
    const wrapper = shallow(<Store />);
    expect(wrapper).toMatchSnapshot();
  });
});
