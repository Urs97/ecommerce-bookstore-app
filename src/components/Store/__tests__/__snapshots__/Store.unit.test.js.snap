// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Store renders Store 1`] = `
<main>
  <PageHeader
    color="blue"
    title="Loading Resources..."
  />
  <section
    className="loader"
  />
</main>
`;

exports[`Store renders empty Store 1`] = `
<main>
  <PageHeader
    color="blue"
    title="Loading Resources..."
  />
  <section
    className="loader"
  />
</main>
`;
